portfolio/src/test/java/com/google/sps/HighestPriorityFirstSchedulerTest.java

```diff
@@ -0,0 +1,309 @@
+    Assert.assertEquals(expected, actual);
+  }
+
+  /** Makes sure that no errors are thrown if working hours is longer than a 24 hour period. */
+  @Test
+  public void workingPeriodExceedsTwentyFourHours() {
+    HighestPriorityFirstScheduler highestPriorityFirstScheduler =
+        new HighestPriorityFirstScheduler();
+
+    Collection<CalendarEvent> events = Arrays.asList();
+    Task task1 =
+        new Task(
+            "Task 1",
+            "First task",
+            SchedulerTestUtil.DURATION_30_MINUTES,
+            SchedulerTestUtil.PRIORITY_ONE);
+    Collection<Task> tasks = Arrays.asList(task1);
+    ScheduledTask scheduledTask1 =
+        new ScheduledTask(
+            task1, SchedulerTestUtil.TIME_0900, SchedulerTestUtil.completelyScheduled);
+
+    Collection<ScheduledTask> actual =
+        highestPriorityFirstScheduler.schedule(
+            events, tasks, SchedulerTestUtil.TIME_0900, SchedulerTestUtil.THREE_DAYS_LATER);
+    Collection<ScheduledTask> expected = Arrays.asList(scheduledTask1);
+    Assert.assertEquals(expected, actual);
+  }
+
+  /** Expected normal behavior if not all tasks can be scheduled. */
+  @Test
+  public void tooManyTasksShouldNotThrowError() {
+    HighestPriorityFirstScheduler highestPriorityFirstScheduler =
+        new HighestPriorityFirstScheduler();
+
+    Collection<CalendarEvent> events =
+        Arrays.asList(
+            new CalendarEvent("Event 1", SchedulerTestUtil.TIME_1000, SchedulerTestUtil.TIME_1800));
+    Task task1 =
+        new Task(
+            "Task 1",
+            "First task",
+            SchedulerTestUtil.DURATION_30_MINUTES,
+            SchedulerTestUtil.PRIORITY_ONE);
+    Task task2 =
+        new Task(
+            "Task 2",
+            "Second task",
+            SchedulerTestUtil.DURATION_30_MINUTES,
+            SchedulerTestUtil.PRIORITY_TWO);
+    Task task3 =
+        new Task(
+            "Task 3",
+            "Third task",
+            SchedulerTestUtil.DURATION_20_MINUTES,
+            SchedulerTestUtil.PRIORITY_FOUR);
+    Task task4 =
+        new Task(
+            "Task 4",
+            "Fourth task",
+            SchedulerTestUtil.DURATION_45_MINUTES,
+            SchedulerTestUtil.PRIORITY_THREE);
+    Collection<Task> tasks = Arrays.asList(task1, task2, task3, task4);
+    ScheduledTask scheduledTask2 =
+        new ScheduledTask(
+            task2, SchedulerTestUtil.TIME_0920, SchedulerTestUtil.completelyScheduled);
+    ScheduledTask scheduledTask3 =
+        new ScheduledTask(
+            task3, SchedulerTestUtil.TIME_0900, SchedulerTestUtil.completelyScheduled);
+
+    Collection<ScheduledTask> actual =
+        highestPriorityFirstScheduler.schedule(
+            events, tasks, SchedulerTestUtil.TIME_0900, SchedulerTestUtil.TIME_1000);
+    Collection<ScheduledTask> expected = Arrays.asList(scheduledTask3, scheduledTask2);
+
+    Assert.assertEquals(expected, actual);
+  }
+}
```

Comment:
I would add a few more tests with tasks of same priority but different duration.

Reply:
Done.